src/Chain.cpp

```diff
@@ -45,7 +45,7 @@ void Chain::Scale(Eigen::MatrixXd& myself, Eigen::MatrixXd& other) const {
 const Eigen::MatrixXd& Chain::marginal() const {
   if (marginal_.size() == 0) {
     // marginal_ hasn't been computed yet, so compute it.
-    marginal_.resize(left_flex() + 1, right_flex() + 1);
+    marginal_.resize(left_flex_ + 1, right_flex_ + 1);
```

Comment:
Purely semantics, no need to call a public member function on a data member within the class itself?